tests/ecosystem/deployment/test_ocs_basic_install.py

```diff
@@ -1,250 +1,12 @@
-import json
 import logging
-import os
-import time
-import yaml
 
-from oc.openshift_ops import OCP
-from ocs.exceptions import CommandFailed, CephHealthException
-from ocs.utils import create_oc_resource, apply_oc_resource
-from ocs.constants import TOP_DIR
-from ocsci.config import RUN, ENV_DATA, DEPLOYMENT
-from ocsci.testlib import deployment, EcosystemTest
-import pytest
-from utility import templating
-from utility.aws import AWS
-from utility.retry import retry
-from utility.utils import run_cmd, get_openshift_installer, get_openshift_client
-from ocs.parallel import parallel
+from ocsci.config import ENV_DATA
+from ocsci.testlib import deployment
+from utility.utils import is_cluster_running
 
 log = logging.getLogger(__name__)
 
 
 @deployment
-class TestDeployment(EcosystemTest):
```

Comment:
How we will deal with the testing of the different deployments?  We will need to test deployment on different platforms and put results to the polarion. So I was thinking about has different test cases for different platform with different polarion_id, and than have some hook to deselect tests which won't be for that specific platform. Can you give me some ideas how we will deal with this? Thanks

Reply:
@petr-balogh I agree. We had some discussion about that last week and we stressed out that we will need a separate set of setup/deployment test cases. @dahorak is going to propose a change of this simple approach when this is needed, so that the 1st version of this setup/teardown fixture could be merged soon.

Basically one would run particular setup, and then run particular tests. The setup fixture will then do nothing when it checks that the cluster is already there.

Reply:
@petr-balogh off the top of my head I think the session fixture added here could become a proxy method that calls out to many other types of deployments. This could be chosen by the user by a new cli option perhaps? We could also use the pytest test collection hook and automatically select tests that would validate the selected deployment method. I believe those tests are what @dahorak was going to propose after this PR to validate the AWS deployment.